doubly_linked_list/doubly_linked_list.py

```diff
@@ -106,7 +106,16 @@ def move_to_end(self, node):
     self.length = self.length - 1
 
   def delete(self, node):
-    pass
+    if self.head is None and self.tail is None:
+      self.head = None
+      self.tail = None
+    else:
+      if node == self.head:
+        self.head = node.next
+      if node == self.tail:
```

Comment:
Based on your condition, if the node is also the head and tail, the value of self.tail will be None since ```node.prev = None```. Therefore, the conditioning of this method is flawed.